tiledb/sm/rest/rest_client.h

```diff
@@ -566,6 +572,10 @@ class RestClient {
    * @return Status
    */
   Status ensure_json_null_delimited_string(Buffer* buffer);
+
+  /** Load all custom headers from the given config.  */
+  static std::unordered_map<std::string, std::string> load_headers(
+      const Config& cfg);
```

Comment:
Returning a map requires copying it. There's no need at all for this since the behavior is being called in (essentially) the constructor.
```suggestion
  void load_headers(const Config& cfg);
```